src/gui/userstatus.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) by Camila <hello@camila.codes>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "userstatus.h"
+#include "account.h"
+#include "accountstate.h"
+#include "networkjobs.h"
+#include "folderman.h"
+#include "creds/abstractcredentials.h"
+#include "theme.h"
+
+#include <QTimer>
+#include <QJsonDocument>
+#include <QJsonObject>
+
+namespace OCC {
+
+Q_LOGGING_CATEGORY(lcUserStatus, "nextcloud.gui.userstatus", QtInfoMsg)
+
+UserStatus::UserStatus(QObject *parent)
+    : QObject(parent)
+    , _message("")
+{
+}
+
+UserStatus::Status UserStatus::stringToEnum(const QString &status) const 
+{
+    // api should return invisible, dnd,... toLower() it is to make sure 
+    // it matches _preDefinedStatus, otherwise the default is online (0)
+    const auto statusEnum = _preDefinedStatus.value(status.isEmpty()? "online" : status.toLower(), 0);
```

Comment:
Shouldn't be 0 but one of the values from the enum (it just so happens to implicitly become Online, but I'd advise against relying on that behavior).

Beside, since _preDefinedStatus content is the same for all UserStatus instances and since stringToEnum won't be called super often, I'd have traded memory use for CPU use and made _preDefinedStatus a local variable instead (and make it `QHash<QString, Status>` then no need for a cast). Has the nice side effect of making the method fully comprehensible when reading it since all the details would be at hand.

Also: "online" should be inside a QStringLiteral and I'd have split the statement in to (one statement for the string representation of the status and one for the lookup).

All together it'd give:
```cpp
const auto statusKey = status.isEmpty() ? QStringLiteral("online") : status.toLower();
return preDefinedStatus.value(statusKey, Online);
```